### ISSUE TITLE
Failing Validation on valid OneOf openapi schema

### ISSUE THREAD

**Issue by thejeff77** [cjbooms/fabrikt#223]
Hey team,

I've been trying to see if this generator will work for us! It seems that the openapi specification files generated by Spring Boot 3's SpringDoc library for our hierarchy classes in kotlin are failing validation via the fabrikt v9 generator. 

Here's an example OpenApi spec file, that passes validation on the [online validator](https://apitools.dev/swagger-parser/online/#), and generates correctly for other clients like the java generator, but fails validation here. 

Thanks for any attention or a fix! It'd be awesome if we could use this.

Command and error message:

`> java -jar fabrikt-9.0.1.jar --output-directory kotlin-fabrikt-client --base-package com.mammalsareawesome --api-file openapi.json --targets CLIENT --targets HTTP_MODELS --validation-library JAKARTA_VALIDATION`

Error:
```
Jul 13, 2023 12:13:34 PM com.cjbooms.fabrikt.cli.CodeGen generate
INFO: Generating code and dumping to kotlin-fabrikt-client/
Exception in thread "main" com.beust.jcommander.ParameterException: Invalid models or api file:
ValidationError(reason='CreateDog' schema contains an invalid combination of properties and `oneOf | anyOf | allOf`. Do not use properties and a combiner at the same level.)
	ValidationError(reason='CreateCow' schema contains an invalid combination of properties and `oneOf | anyOf | allOf`. Do not use properties and a combiner at the same level.)
	at com.cjbooms.fabrikt.model.SourceApi.<init>(SourceApi.kt:40)
	at com.cjbooms.fabrikt.model.SourceApi$Companion.create(SourceApi.kt:31)
	at com.cjbooms.fabrikt.cli.CodeGen.generate(CodeGen.kt:62)
	at com.cjbooms.fabrikt.cli.CodeGen.main(CodeGen.kt:22)
```

OpenApi Spec File:

```
{
    "openapi": "3.0.1",
    "info": {
        "title": "Mammal Microservice",
        "version": "0.0.1"
    },
    "externalDocs": {
        "description": "github",
        "url": "https://github.com/fakenamespace/mammal-service"
    },
    "servers": [
        {
            "url": "http://localhost:8080",
            "description": "Generated server url"
        }
    ],
    "security": [
        {
            "api-key": []
        }
    ],
    "paths": {
        "/api/v1/mammals": {
            "post": {
                "tags": [
                    "mammal-controller"
                ],
                "summary": "Create Mammal APIs.",
                "operationId": "createMammal",
                "requestBody": {
                    "content": {
                        "application/json": {
                            "schema": {
                                "$ref": "#/components/schemas/CreateMammalRequest"
                            }
                        }
                    },
                    "required": true
                },
                "responses": {
                    "201": {
                        "description": "Created",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/MammalCreated"
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad Request",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/MammalCreateFailure"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "MammalType": {
                "type": "string",
                "enum": [
                    "DOG",
                    "COW",
                ]
            },
            "CreateDog": {
                "required": [
                    "mammalType",
                    "doesBark",
                ],
                "type": "object",
                "allOf": [
                    {
                        "$ref": "#/components/schemas/CreateMammal"
                    },
                    {
                        "type": "object",
                        "properties": {
                            "doesBark": {
                                "type": "boolean"
                            }
                        }
                    }
                ]
            },
            "CreateCow": {
                "required": [
                    "mammalType",
                    "doesTip",
                ],
                "type": "object",
                "allOf": [
                    {
                        "$ref": "#/components/schemas/CreateMammal"
                    },
                    {
                        "type": "object",
                        "properties": {
                            "doesTip": {
                                "type": "boolean",
                                "description": "If you can push it at night and it tips over."
                            }
                        }
                    }
                ]
            },
            "CreateMammal": {
                "required": [
                    "mammalType",
                ],
                "type": "object",
                "properties": {
                    "mammalType": {
                        "$ref": "#/components/schemas/MammalType"
                    }
                },
                "description": "Mammal Creation Parent Object.",
                "discriminator": {
                    "propertyName": "mammalType"
                }
            },
            "CreateMammalRequest": {
                "required": [
                    "createMammal",
                    "requestId",
                    "skipLockValidation"
                ],
                "type": "object",
                "properties": {
                    "createMammal": {
                        "oneOf": [
                            {
                                "$ref": "#/components/schemas/CreateDog"
                            },
                            {
                                "$ref": "#/components/schemas/CreateCow"
                            },
                        ]
                    },
                    "requestId": {
                        "type": "string",
                        "format": "uuid"
                    }
                }
            },
            "MammalCreateFailure": {
                "required": [
                    "error",
                ],
                "type": "object",
                "properties": {
                    "error": {
                        "type": "string"
                    }
                }
            },
            "MammalCreated": {
                "required": [
                    "message",
                ],
                "type": "object",
                "properties": {
                    "message": {
                        "type": "string"
                    }
                }
            }
        },
        "securitySchemes": {
            "api-key": {
                "type": "apiKey",
                "name": "api-key",
                "in": "header"
            }
        }
    }
}
```

**Comment by cjbooms** [comment]
Assuming the request here is to permit schemas that declare `oneOf`, `anyOf`, or `allOf`, and also be marked as `type: object` I think we can probably make that work.

However, the example API spec you provided is not valid in the following ways that I am ignoring:
 - Arrays with trailing commas
 - discriminator does not include a mapping to different correct types
 
 See linked PR branch with how I think you want it to look 
https://github.com/cjbooms/fabrikt/pull/224/files#diff-28bb2f762d6ea8c25d54e8615124f4f096c715b2889504d9b06e33c3536aefb4

**Comment by thejeff77** [comment]
@cjbooms thank you! This does look like the intended spec. Somehow the openapi validator didn't flag the lack of discriminator mapping for me, but we can easily add it.

Thanks for the fix and turn around of this. I'm eager to give the generator another shot.

**Comment by cjbooms** [comment]
It's not quite fixed yet, that test fails. Somewhere the json sub type annotation generation will need a fix

**Comment by cjbooms** [comment]
Hopefully fixed in 10.1.0. Please confirm when you can
